modules/apps/frontend-taglib/frontend-taglib-clay/src/main/resources/META-INF/resources/UserCard.js

```diff
@@ -0,0 +1,79 @@
+/**
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+import {ClayCardWithUser} from '@clayui/card';
+import React, {useState} from 'react';
+
+const getDataAttributes = (data) => {
+	return data
+		? Object.entries(data).reduce((acc, [key, value]) => {
+				acc[`data-${key}`] = value;
+
+				return acc;
+		  }, {})
+		: {};
+};
+
+export default function UserCard({
+	actions = [],
+	componentId: _componentId,
+	cssClass,
+	inputName,
+	inputValue,
+	labels = [],
+	locale: _locale,
+	portletId: _portletId,
+	portletNamespace: _portletNamespace,
+	selected: initialSelected,
+	selectable,
+	...otherProps
+}) {
+	const [selected, setSelected] = useState(initialSelected);
+
+	return (
+		<ClayCardWithUser
+			actions={actions?.map(({data, ...rest}) => {
+				const dataAttributes = getDataAttributes(data);
+
+				return {
+					...dataAttributes,
+					...rest,
+				};
+			})}
```

Comment:
This is unusual :thinking: . Are you sure this is the correct way to handle data attributes? In `HorizontalCard`, we are setting data attributes to container. Then, actions have code that read it from there and do custom stuff, like open modal. I would imagine `UserCard` would do something similar.

If this is your case, then yeah, [setting `data` in backend](https://github.com/liferay-frontend/liferay-portal/pull/449/commits/b0c7c00d25aae3096f392f6e689cea7ae1973dbf#diff-a45f3b1f2a6d6fad307b929f02e524e2759fe240dd3bf32d0b73dd5b88035225R42-R50) is not an option here.

Reply:
This is `data` comming` from the actions, not from the component

Reply:
Same use case here: https://github.com/liferay/liferay-portal/blob/a89f01c3f01c27369f271e56c51d76040823d489/modules/apps/frontend-taglib/frontend-taglib-clay/src/main/resources/META-INF/resources/DropdownMenu.js#L40

Reply:
Oh right, sorry, I misunderstood the snippet. All good here :+1: